src/transformers/adapters/peft_mixin.py

```diff
@@ -0,0 +1,333 @@
+
+
+class PeftAdapterMixin:
+    """
+    A class containing all functions for loading and using adapters weights that are supported in PEFT library.
+    Currently supported PEFT methods are all non-prefix tuning methods.
+
+    With this mixin, if the correct PEFT version is installed, it is possible to:
+
+    - Load an adapter stored on a local path or in a remote Hub repository, and inject it in the model
+    - Attach new adapters in the model and train them with Trainer or by your own.
+    - Attach multiple adapters and iteratively activate / deactivate them
+    - Activate / deactivate all adapters from the model.
+    - Get the `state_dict` of the active adapter.
+    """
+
+    _hf_peft_config_loaded = False
+
+    def load_adapter(
+        self,
+        peft_model_id: str,
+        adapter_name: Optional[str] = None,
+        revision: Optional[str] = None,
+        use_auth_token: Optional[str] = None,
+        commit_hash: Optional[str] = None,
+        device_map: Optional[str] = "auto",
+        max_memory: Optional[str] = None,
+        offload_dir: Optional[str] = None,
+        offload_index: Optional[int] = None,
+    ) -> None:
+        """
+        Load adapter weights from file or remote Hub folder. Requires peft as a backend to load the adapter weights.
+
+        Args:
+            peft_model_id (`str`):
+                The identifier of the model to look for on the Hub, or a local path to the saved adapter config file
+                and adapter weights.
+            adapter_name (`str`, `optional`):
+                The adapter name to use. If not set, will use the default adapter.
+            revision (`str`, `optional`):
+                revision argument to be passed to `hf_hub_download` method from `huggingface_hub`.
+            use_auth_token (`str`, `optional`):
+                use_auth_token argument to be passed to `hf_hub_download` method from `huggingface_hub`.
+            commit_hash (`str`, `optional`):
+                commit_hash argument to be passed to `hf_hub_download` method from `huggingface_hub`.
+            device_map (`str`, `optional`):
+                The device map to use for the adapter. Used only in the case the model is offloaded into the CPU to
+                correctly re-dispatch the model together with the adapter weights.
+            max_memory (`str`, `optional`):
+                The maximum memory to use for the adapter. Used only in the case the model is offloaded into the CPU
+                for the same reasons stated above.
+            offload_dir (`str`, `optional`):
+                The directory to use for offloading the model. Used only in the case the model is offloaded into the
+                CPU.
+            offload_index (`int`, `optional`):
+                `offload_index` argument to be passed to `accelerate.dispatch_model` method.
+        """
+        check_peft_version(min_version="0.4.0")
+
+        adapter_name = adapter_name or "default"
```

Comment:
We don't use magic conversion to bool in Transformers, so you should explicitly test here:
```suggestion
        adapter_name = adapter_name if adapater_name is not None else "default"
```